src/envs/pybullet_blocks.py

```diff
@@ -0,0 +1,728 @@
+            current, target = get_current_and_target_pose(
+                state, objects, params)
+            return self._get_move_toward_waypoint_action(current, target)
+
+        def _terminal(state: State, memory: Dict, objects: Sequence[Object],
+                      params: Array) -> bool:
+            del memory  # unused
+            current, target = get_current_and_target_pose(
+                state, objects, params)
+            dist = np.sum(np.subtract(current, target)**2)
+            return dist < self._move_to_pose_tol
+
+        return ParameterizedOption(name,
+                                   types=types,
+                                   params_space=params_space,
+                                   policy=_policy,
+                                   initiable=lambda _1, _2, _3, _4: True,
+                                   terminal=_terminal)
+
+    def _move_relative_to_block(self, name: str,
+                                rel_or_abs_target_pose: Pose3D,
+                                rel_or_abs: Tuple[str, str,
+                                                  str], types: Sequence[Type],
+                                params_space: Box) -> ParameterizedOption:
+        """Creates a ParameterizedOption for moving to a target pose relative
+        to a block.
+
+        Each of the three dimensions of the target pose can be specified
+        relatively or absolutely. The rel_or_abs argument indicates
+        whether each dimension is relative ("rel") or absolute ("abs").
+        """
+        assert types == [self._robot_type, self._block_type]
+
+        def _get_current_and_target_pose(
+                state: State, objects: Sequence[Object],
+                params: Array) -> Tuple[Pose3D, Pose3D]:
+            del params  # not used
+            robot, block = objects
+            current_pose = (state[robot][0], state[robot][1], state[robot][2])
+            block_pose = (state[block][0], state[block][1], state[block][2])
+            target_pose = self._convert_rel_abs_to_abs(block_pose,
+                                                       rel_or_abs_target_pose,
+                                                       rel_or_abs)
+            return current_pose, target_pose
+
+        return self._move_to_pose(name, types, params_space,
+                                  _get_current_and_target_pose)
+
+    def _change_grippers(self, name: str, target_val: float,
+                         types: Sequence[Type],
+                         params_space: Box) -> ParameterizedOption:
+        """Creates a ParameterizedOption for changing the robot grippers."""
+
+        assert types[0] == self._robot_type
+
+        def _policy(state: State, memory: Dict, objects: Sequence[Object],
+                    params: Array) -> Action:
+            del memory, objects, params  # unused
+            current_val = state.get(self._robot, "fingers")
+            gripper_action = target_val - current_val
```

Comment:
if the gripper_action really is just this delta, instead of a discrete -1/0/1, then i'm now especially confused by the `if finger_action == 0:` line in step().

Reply:
it's zero for other options, not this one. but I changed that line anyway